pkgs/by-name/pr/private-gpt/package.nix

```diff
@@ -0,0 +1,13 @@
+{ python3Packages
+, makeBinaryWrapper
+}:
+
+python3Packages.toPythonApplication (python3Packages.private-gpt.overrideAttrs (oldAttrs: {
+  nativeBuildInputs = oldAttrs.nativeBuildInputs ++ [ makeBinaryWrapper ];
+
+  postInstall = ''
+    makeWrapper ${python3Packages.python.interpreter} $out/bin/private-gpt \
+      --prefix PYTHONPATH : "$PYTHONPATH" \
+      --add-flags "-m private_gpt"
```

Comment:
Do you think upstream would accept a PR wrapping their `uvicorn.run(...)` in:

```
def main():
  ...

if __name__ == "__main__":
  main()
```

so that you could instead specify `...:main` in pyproject.toml's `scripts`?

It's preferable to avoid setting `PYTHONPATH` when possible

Reply:
I don't use python usually and I'm not acquainted with its ecosystem. So the least help is very welcome .